streams/src/test/java/org/apache/kafka/streams/state/internals/ListValueStoreTest.java

```diff
@@ -178,26 +181,46 @@ public void shouldGetAllReturnTimestampOrderedRecords() {
 
     @Test
     public void shouldAllowDeleteWhileIterateRecords() {
-        listStore.put(0, "zero1");
-        listStore.put(0, "zero2");
-        listStore.put(1, "one");
+        final Random rand = new Random();
+        int count = 0;
+        for (int i = 0; i < 10000; i++) {
+            listStore.put(i, "zero" + i);
+            count++;
+
+            while (rand.nextBoolean()) {
+                listStore.put(i, "zero" + i);
+                count++;
+            }
+        }
 
-        final KeyValue<Integer, String> zero1 = KeyValue.pair(0, "zero1");
-        final KeyValue<Integer, String> zero2 = KeyValue.pair(0, "zero2");
-        final KeyValue<Integer, String> one = KeyValue.pair(1, "one");
+        final int size = toList(listStore.all()).size();
```

Comment:
`listStore.all()` will return a `KeyValueIterator`, which should be explicitly closed. I think we can re-use the iterator below this line, which is closed at the end.